pyop2/mpi.py

```diff
@@ -278,15 +343,18 @@ def dup_comm(comm_in):
         # Name
         internal_comm.Set_name(f"{comm_in.name or comm_in.py2f()}_DUP")
         # Refcount
-        internal_comm.Set_attr(refcount_keyval, [0])
+        internal_comm.Set_attr(refcount_keyval, [1])
         incref(internal_comm)
         # Remember we need to destroy it.
-        dupped_comms.append(internal_comm)
+        debug(f"Appending comm {internal_comm.name} to list of known comms")
+        cidx = _COMM_CIDX.__next__()
```

Comment:
You can just call `next(_COMM_CIDX)`

Reply:
:facepalm:  of course I can, I also forgot to lint